### PR TITLE
Invert apiVersion and kind in docs

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/kubernetes-sigs/gateway-api/issues/810

Done with intellij find+replace on regex `kind: (.*)\napiVersion: (.*)`
to `apiVersion: $1\nkind: $2`

### CODE REVIEW

**Comment by robscott** [comment]
Thanks!

/ok-to-test
/lgtm

File: examples/v1alpha2/basic-http.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
-kind: GatewayClass
-apiVersion: gateway.networking.k8s.io/v1alpha2
+apiVersion: GatewayClass
+kind: gateway.networking.k8s.io/v1alpha2
```
Comment (hbagdi) [inline examples/v1alpha2/basic-http.yaml:2]: something went wrong here. Applicable to multiple resources.

Comment (howardjohn) [reply examples/v1alpha2/basic-http.yaml]: I knew something looked off :slightly_smiling_face: should be fixed now

**Comment by hbagdi** [comment]
Thanks for improving the consistency @howardjohn.
My OCD is indebted to you.

/approve
/lgtm

**Pull request merged**